src/main/java/org/opensearch/knn/index/codec/nativeindex/IndexBuildHelper.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.knn.index.codec.nativeindex;
+
+import lombok.experimental.UtilityClass;
+import org.opensearch.knn.index.codec.nativeindex.model.BuildIndexParams;
+import org.opensearch.knn.index.quantizationService.QuantizationService;
+import org.opensearch.knn.index.vectorvalues.KNNVectorValues;
+import org.opensearch.knn.quantization.models.quantizationOutput.QuantizationOutput;
+import org.opensearch.knn.quantization.models.quantizationState.QuantizationState;
+
+import java.io.IOException;
+
+@UtilityClass
+class IndexBuildHelper {
+
+    /**
+     * Processes and returns the vector based on whether quantization is applied or not.
+     *
+     * @param knnVectorValues the KNN vector values to be processed.
+     * @param indexBuildSetup the setup containing quantization state and output, along with other parameters.
+     * @return the processed vector, either quantized or original.
+     * @throws IOException if an I/O error occurs during processing.
+     */
+    static Object processAndReturnVector(KNNVectorValues<?> knnVectorValues, IndexBuildSetup indexBuildSetup) throws IOException {
+        QuantizationService quantizationService = QuantizationService.getInstance();
+        if (indexBuildSetup.getQuantizationState() != null && indexBuildSetup.getQuantizationOutput() != null) {
+            quantizationService.quantize(
+                indexBuildSetup.getQuantizationState(),
+                knnVectorValues.getVector(),
+                indexBuildSetup.getQuantizationOutput()
+            );
+            return indexBuildSetup.getQuantizationOutput().getQuantizedVector();
+        } else {
+            return knnVectorValues.conditionalCloneVector();
+        }
+    }
+
+    /**
+     * Prepares the quantization setup including bytes per vector and dimensions.
+     *
+     * @param knnVectorValues the KNN vector values.
+     * @param indexInfo the index build parameters.
+     * @return an instance of QuantizationSetup containing relevant information.
+     */
+    static IndexBuildSetup prepareIndexBuild(KNNVectorValues<?> knnVectorValues, BuildIndexParams indexInfo) {
```

Comment:
This function can be easily moved to IndexBuildSetup class, and renamed as `static IndexBuildSetup build(KNNVectorValues<?> knnVectorValues, BuildIndexParams indexInfo)`.

and can be used as way to build the IndexBuildSetup object.

Reply:
I would my POJO Simple.  and act as Simple Model Class. So I would using KISS here.

Reply:
if that is the case then Builder classes in the Builder patter will be separate from POJO. Moving this function in POJO class has it advantages in terms of restricting the usage of the class like making constructor private, any user of this class knowing how to create the `IndexBuildSetup` POJO.

Reply:
This is called Builder design pattern which tell if you have any complex Pojo , then how to build it , or for any normal pojo also ..It would be ...I would prefer keeping Model class Simple.

Reply:
I don't agree here. But neither I am blocking the PR because of this. So, leaving it upto you.

Reply:
Thanks